### DISCUSSION TITLE
Way to get caller client ip address in RouterRequest

### DISCUSSION THREAD

**Discussion by golfingal72** [apollographql/router#1212] [Q&A] [answered]
Is there a way to get the callers client ip address from the RouterRequest???

**Comment by o0Ignition0o** [comment]
This is a great idea!

As part of our efforts to stabilize the extensions api, we're going to think of a way to expose this, possibly using [ConnectInfo](https://docs.rs/axum/latest/axum/extract/struct.ConnectInfo.html).

**Reply by golfingal72** [reply]
@o0Ignition0o Even if your shadow on the http::compat::Request you could just store the ip address to expose it.. i think there are methods in the originial object to show the caller ip already

**Reply by o0Ignition0o** [reply]
I couldn't find any when I looked, but I haven't dug that much. That being said if there's a way to extract the client ip from an http request you can [try to transform it for testing purposes](https://github.com/apollographql/router/blob/ac55ed6d8bd73e2beef89ab48affb3b06668d50e/apollo-router-core/src/services/http_compat.rs#L180)

**Comment by golfingal72** [comment]
The reason i ask is because I have written custom logging to "mimick" a tomcat access log that we are consuming in datadog.. the first  part of the output is the caller ip address.. you can see the progressions to downstream services

`0:0:0:0:0:0:0:1 - - [10/Jun/2022:10:31:48.780 -0400] "POST /localhost:9080/api/v4/graphql/shpst_favoritesSearch?cashback=0&inStock=true&limit=2&offset=0&userId=23425584 HTTP/1.1" 200 1297 "ff99705d-ee79-48ee-ba88-aa745509bae3" "1654871508548051000" "GraphQLRestGateway#@US25020249" 232

0:0:0:0:0:0:0:1 - - [10/Jun/2022:10:31:48.957 -0400] "POST /localhost:9090/api/v4/graphql/shpst_favoritesSearch/entity/Product?id=920196886&curatorUserId=23425584&productIndexTs=1653464915000&favCount=21&imageId=d8c670167646f48c7dbb22bbb211d0cb&userHasFav=true&queryLocale=en_US&id=818503709&curatorUserId=23425584&productIndexTs=1653463312000&favCount=30&imageId=48adeba416ae0fedb3a96df3014f60a8&userHasFav=true&queryLocale=en_US HTTP/1.1" 200 7825 "ff99705d-ee79-48ee-ba88-aa745509bae3" "1654871508781815000" "GraphQLRestGateway#@US25020249" 175

0:0:0:0:0:0:0:1 - - [10/Jun/2022:10:31:48.984 -0400] "POST /localhost:9095/api/v4/graphql/shpst_favoritesSearch/entity/MetadataUser?id=23425584 HTTP/1.1" 200 134 "ff99705d-ee79-48ee-ba88-aa745509bae3" "1654871508781851000" "GraphQLRestGateway#@US25020249" 202

0:0:0:0:0:0:0:1 - - [10/Jun/2022:10:31:48.988 -0400] "POST /shpst_favoritesSearch?cashback=0&inStock=true&limit=2&offset=0&userId=23425584 HTTP/1.1" 200 8695 "ff99705d-ee79-48ee-ba88-aa745509bae3" "1654871508487110000" "GraphQLRestGateway#@US25020249" 501`

**Reply by o0Ignition0o** [reply]
I see, that makes sense indeed! As an aside, is there a reason why the scheme isn't displayed in the log entries ?

**Reply by golfingal72** [reply]
i am not following...  the entries contain the query operation_name and in our system, that is the schema query with a prefix

**Reply by o0Ignition0o** [reply]
By scheme I meant « http » or « https »; the logs seem to indicate /localhost:9095/

**Reply by o0Ignition0o** [reply]
I might have missread it since it could be part of the path though!

**Reply by golfingal72** [reply]
typical tomcat logs (which i am mimicking), never have the http:// part on it :)

**Comment by golfingal72** [comment, marked as answer]
answered in other issue about getting server name
